### PR TITLE
[9.x] Allow overriding transport type on Mailgun transporter

### PR DESCRIPTION
**Pull request by jnoordsij**
In #41255 the Mailgun transport type was changed from `api` to `https`. This breaks my application, as I add some custom headers which were working fine with the API transporter, but are ignored on the messages.mime endpoint. This MR allows overriding the transport type you want to use for Mailgun, while leaving the default option unchanged.

If wanted, I can also provide a PR to add this config key to the `laravel/laravel` repo.

### CODE REVIEW

**Comment by driesvints** [comment]
Thanks @jnoordsij. I still feel this change was warranted as we used the `/message.mime` endpoint in Laravel v8.

Just to be clear: you weren't using these custom headers in Laravel 8, correct?

File: src/Illuminate/Mail/MailManager.php [hunk 1]
```diff
@@ -281,7 +281,7 @@ protected function createMailgunTransport(array $config)
         }
 
         return $factory->create(new Dsn(
-            'mailgun+https',
+            'mailgun+'.($config['transport_type'] ?? 'https'),
```
Comment (driesvints) [inline src/Illuminate/Mail/MailManager.php:284]: This should be `scheme` to match Symfony's terminology.

```suggestion
            'mailgun+'.($config['scheme'] ?? 'https'),
```

**Review by driesvints — CHANGES_REQUESTED** [review]

**Comment by driesvints** [comment]
This indeed also needs a PR to `laravel/laravel`.

**Comment by jnoordsij** [comment]
> Thanks @jnoordsij. I still feel this change was warranted as we used the `/message.mime` endpoint in Laravel v8.

Yeah, the change itself seems logical, its just even more flexible this way I feel.

> Just to be clear: you weren't using these custom headers in Laravel 8, correct?

I was using them in Laravel 8, but had to rewrite the implementation in Laravel 9. I used to extend the `Illuminate\Mail\Transport\MailgunTransport` to alter the HTTP payload, but this transport was removed. As this was removed in Laravel 9, I had to rewrite things once already. Adding text headers for the testmode in Laravel 9 worked fine with the API endpoint, but I can't seem to get it working again with the messages.mime endpoint.

For reference, I tried to obtain the request body before sending it in both my Laravel 8 and Laravel 9 apps, and they seem to be quite different:

*Laravel 8*
```
--553506d1f480caca62c55a8678ed647e36bcd10a\r\n
Content-Disposition: form-data; name="to"\r\n
Content-Length: 20\r\n
\r\n
mailgun@example.com\r\n
--553506d1f480caca62c55a8678ed647e36bcd10a\r\n
Content-Disposition: form-data; name="message"; filename="message.mime"\r\n
Content-Length: 324\r\n
\r\n
Message-ID: <f1c214aed4da7e9153a3f9430c02f62d@swift.generated>\r\n
Date: Thu, 03 Mar 2022 10:41:04 +0100\r\n
Subject: Test message\r\n
From: Example <test@example.com>\r\n
To: mailgun@example.com\r\n
MIME-Version: 1.0\r\n
Content-Type: text/html; charset=utf-8\r\n
Content-Transfer-Encoding: quoted-printable\r\n
\r\n
<p>Hello world!</p>\r\n
--553506d1f480caca62c55a8678ed647e36bcd10a\r\n
Content-Disposition: form-data; name="o:testmode"\r\n
Content-Length: 3\r\n
\r\n
yes\r\n
--553506d1f480caca62c55a8678ed647e36bcd10a\r\n
Content-Disposition: form-data; name="o:tag"\r\n
Content-Length: 26\r\n
\r\n
App Name (local)\r\n
--553506d1f480caca62c55a8678ed647e36bcd10a--\r\n
"""
```
*Laravel 9*
```
'body' => 'From: Example <test@example.com>
To: mailgun@example.com
Subject: Test message
o:testmode: yes
X-Mailgun-Tag: App Name (local)
Message-ID: <628fcc053ba6161ff327e492274005d6@example.com>
MIME-Version: 1.0
Date: Thu, 03 Mar 2022 10:57:09 +0100
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: quoted-printable

<p>Hello world!</p>'
```

**Comment by driesvints** [comment]
@jnoordsij all of that is being handled by Symfony Mailer so there's nothing we can really do there.

**Comment by jnoordsij** [comment]
> @jnoordsij all of that is being handled by Symfony Mailer so there's nothing we can really do there.

Yeah I figured, hence having the option to just stick with the `api` transporter for myself seems to be the best solution for now.

**Review by driesvints — APPROVED** [review]

**Pull request merged**